### DISCUSSION TITLE
SkipWhile not working

### DISCUSSION THREAD

**Discussion by petersaj** [bonsai-rx/bonsai#1703] [General]
Hi - I have a question about the SkipWhile: the documentation seems very clear, but this node doesn't seem to perform as expected. Here's an example below, where the conditions in SkipWhile are fullfilled (workflowoutput is true), but the SkipWhile node continues to pass elements. Am I using this incorrectly?

![image](https://github.com/bonsai-rx/bonsai/assets/17985056/48e27f14-4915-4f31-b593-949f98056eeb)

**Comment by bruno-f-cruz** [comment]
Please attach the workflow. Thanks

**Comment by petersaj** [comment]
Sure here's a little example workflow (added .txt extension to upload): there an accumulator with +1/-1 on up/down keypress, the ending SkipWhile node should filter out elements whenever they're >5, but instead it passes all elements through

(I did end up solving my original issue with a Scan, but good to know about this for future reference)

[skipwhile_test.bonsai.txt](https://github.com/bonsai-rx/bonsai/files/14461950/skipwhile_test.bonsai.txt)

**Reply by bruno-f-cruz** [reply]
Not sure I see the problem. The operator states: "... bypasses elements in an observable sequence as long as the condition specified by the encapsulated workflow is true.". Is it the case you thought the boolean logic was reversed?

**Reply by petersaj** [reply]
Ah I see, I misinterpreted "SkipWhile will start emitting all elements from the source sequence" - I thought this was dynamic (would turn on and off depending on condition) rather than on a switch (off at start, then on permanently once the condition is true), thanks.

**Reply by bruno-f-cruz** [reply]
That behavior you describing is implemented using the Condition operator.

**Reply by petersaj** [reply]
Oh perfect, thanks! I didn't realize condition worked like that.
